### PR TITLE
enhancement: don't allow wallet names to start with '@' #3202

### PR DESCRIPTION
**Pull request by jeivardan**
resolves [#3202](https://github.com/coral-xyz/backpack/issues/3202) in both create and rename wallet

renameWallet

createWallet

### CODE REVIEW

File: packages/app-extension/src/components/Unlocked/Settings/YourAccount/EditWallets/RenameWallet.tsx [hunk 1]
```diff
@@ -24,6 +27,20 @@ export const RenameWallet: React.FC<{ publicKey: string; name: string }> = ({
     nav.setOptions({ headerTitle: "Rename Wallet" });
   }, [nav]);
 
+  useEffect(() => {
+    //regex check to not allow wallet names to start with '@'
+    let regExp = new RegExp("^(?!@.*$).*");
+    if (!regExp.test(walletName)) {
```
Comment (hkirat) [inline packages/app-extension/src/components/Unlocked/Settings/YourAccount/EditWallets/RenameWallet.tsx:33]: Seems like code is duplicated in two files. Can u move it to either common or some place inside `app-extension`?
Other than that looks good

**Review by hkirat — CHANGES_REQUESTED** [review]

File: packages/app-extension/src/components/Unlocked/Settings/YourAccount/EditWallets/RenameWallet.tsx (hunk 1 above)
Comment (jeivardan) [reply packages/app-extension/src/components/Unlocked/Settings/YourAccount/EditWallets/RenameWallet.tsx:33]: Sure will look into it and will make the changes

Comment (jeivardan) [reply packages/app-extension/src/components/Unlocked/Settings/YourAccount/EditWallets/RenameWallet.tsx]: @hkirat I moved the code to common

**Review by jeivardan — COMMENTED** [review]
@hkirat I moved the code to common

File: packages/common/src/utils.ts [hunk 2]
```diff
@@ -63,6 +63,17 @@ export function isServiceWorker(): boolean {
...
 }
 
+/**
+ * Validate a WalletName
+ * @param walletName
+ *
+ */
+export function isValidWalletName(walletName: string): boolean {
+  //regex check to not allow wallet names to start with '@'
+  let regExp = new RegExp("^(?!@.*$).*");
```
[the remark this answers is not in the document]
Comment (jeivardan) [reply packages/common/src/utils.ts]: Yep it is too generic and I've made the changes

File: packages/app-extension/src/components/Unlocked/Settings/YourAccount/EditWallets/RenameWallet.tsx [hunk 3]
```diff
@@ -24,6 +28,18 @@ export const RenameWallet: React.FC<{ publicKey: string; name: string }> = ({
...
 
+  useEffect(() => {
+    if (!isValidWalletName(walletName)) {
+      setError(true);
+      setErrorMessage("Wallet name should not begin with '@'");
+      setIsPrimaryDisabled(true);
+    } else {
+      setError(false);
+      setErrorMessage("");
+      setIsPrimaryDisabled(walletName.trim() === "" ? true : false);
```
[the remark this answers is not in the document]
Comment (jeivardan) [reply packages/app-extension/src/components/Unlocked/Settings/YourAccount/EditWallets/RenameWallet.tsx]: Yeah! Simplified it. Thanks for the suggestion.

**Pull request closed without merging**